test/unit-tests/utils/number.test.js

```diff
@@ -505,83 +505,97 @@ describe('number', function () {
+      assert.strictEqual(nearlyEqual(0, -0, relTol, absTol), true)
+      assert.strictEqual(nearlyEqual(0, 1.2, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(0, 1e30, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(0, 1e-30, relTol, absTol), true) // FIXME: why is this true?
     })
 
     it('should compare with Infinity', function () {
-      const epsilon = 1e-3
-
-      assert.strictEqual(nearlyEqual(1.2, Infinity, epsilon), false)
-      assert.strictEqual(nearlyEqual(Infinity, 1.2, epsilon), false)
-      assert.strictEqual(nearlyEqual(Infinity, Infinity, epsilon), true)
-      assert.strictEqual(nearlyEqual(Infinity, -Infinity, epsilon), false)
-      assert.strictEqual(nearlyEqual(-Infinity, Infinity, epsilon), false)
-      assert.strictEqual(nearlyEqual(-Infinity, -Infinity, epsilon), true)
+      const relTol = 1e-3
+      const absTol = 1e-6
+
+      assert.strictEqual(nearlyEqual(1.2, Infinity, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(Infinity, 1.2, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(Infinity, Infinity, relTol, absTol), true)
+      assert.strictEqual(nearlyEqual(Infinity, -Infinity, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(-Infinity, Infinity, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(-Infinity, -Infinity, relTol, absTol), true)
     })
 
     it('should compare with NaN', function () {
-      const epsilon = 1e-3
-      assert.strictEqual(nearlyEqual(1.2, NaN, epsilon), false)
-      assert.strictEqual(nearlyEqual(NaN, 1.2, epsilon), false)
-      assert.strictEqual(nearlyEqual(NaN, NaN, epsilon), false)
+      const relTol = 1e-3
+      const absTol = 1e-6
+      assert.strictEqual(nearlyEqual(1.2, NaN, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(NaN, 1.2, relTol, absTol), false)
+      assert.strictEqual(nearlyEqual(NaN, NaN, relTol, absTol), false)
     })
 
-    it('should do exact comparison when epsilon is null or undefined', function () {
+    it('should use default values when absTol and relTol are undefined', function () {
       assert.strictEqual(nearlyEqual(1.2, 1.2), true)
-      assert.strictEqual(nearlyEqual(1.2, 1.2, null), true)
+      assert.strictEqual(nearlyEqual(1.2, 1.2, undefined), true)
+      assert.strictEqual(nearlyEqual(1.2, 1.2, undefined, undefined), true)
 
-      assert.strictEqual(nearlyEqual(0.1 + 0.2, 0.3), false)
-      assert.strictEqual(nearlyEqual(0.1 + 0.2, 0.3, null), false)
+      assert.strictEqual(nearlyEqual(0.1 + 0.2, 0.3), true)
+      assert.strictEqual(nearlyEqual(0.1 + 0.2, 0.3, undefined), true)
+      assert.strictEqual(nearlyEqual(0.1 + 0.2, 0.3, undefined, undefined), true)
+
+      assert.strictEqual(nearlyEqual(1.2 + 1e-7, 1.2), false)
+      assert.strictEqual(nearlyEqual(1.2 + 1e-7, 1.2, undefined), false)
+      assert.strictEqual(nearlyEqual(1.2 + 1e-7, 1.2, undefined, undefined), false)
```

Comment:
Can you add a new unit test that validates the backward compatibility, using the deprecated `epsilon`?

Reply:
Yes of course,

I did include them on `config.test.js` but found an issue with console.warn that I don't know how to test and also breaks the pattern of the tests that now looks like this.

```
.....................................Warning: The configuration option "epsilon" is deprecated. Use "relTol" and "absTol" instead.
.....................,,,,,.......................
  ......................................................................................
```

This is my error but I don't know how to fix.

Reply:
Ah, yes. We can do two things:

1. Integrate and use some spy library for mocha, like [Sinon](https://sinonjs.org/releases/latest/spies/)
2. Temporarily replacing `console.warn` with our own custom function, and after the test restore the original `console.warn`

Option 1 would be best, but only if it is simple to setup, otherwise option 2 is fine with me. Option 2 is a bit ugly: when the test throws an exception the original console.warn is not restored, therefore it's not my first choice.

Reply:
Thanks!

I will research sinon and report back.

Reply:
Hi, it seems to be working now, please review.

Reply:
Nice job, thanks!